DirectXTex/Shaders/BC6HEncode.hlsl

```diff
@@ -979,14 +979,15 @@ void EncodeBlockCS(uint GI : SV_GroupIndex, uint3 groupID : SV_GroupID)
     }
 }
 
-uint float2half1( float f )
+#ifdef EMULATE_F16C
```

Comment:
The original implementation emulated the ``f32tof16`` intrinsics directly. While the HLSL compiler will emulate this for you on SM 4.x, it turns out the emulation is slower than the original code on 4.x targets. It is faster to use the intrinsic on SM 5.0 targets.

Therefore, the logic here in this change is to use the original emulation code for SM 4.0 and the intrinsics for SM 5.0.